packages/Python/lldbsuite/test/lang/swift/closure_shortcuts/main.swift

```diff
@@ -8,7 +8,9 @@ func main() -> Int {
               //%self.expect('frame var $1', substrs=['foo'])
   )
 
-  return 0
+  var tinky = [1,2].map({$0 * 2}) //%self.expect('expr [1,2].map({$0 * 2})', substrs=['[0] = 2', '[1] = 4'])
```

Comment:
Might be better to break this out into multiple lines, so you can be sure you're testing this from inside the closure. Or, actually, this should just be part of the previous test, no?

Reply:
honestly, this is probably better rewritten as 

```
var tinky = [1,2]
//%self.expect('expr tinky.map({$0 * 2}, substr=[...]
```

Reply:
I thought Davide was being clever and getting two test for the price of one.  After all, the breakpoint on the line will actually get two locations, one in the main function, and one in the closure.  So as written the test will actually make sure we can use $0 in a being defined closure both in a closure context and out of it.  Since the two had different failure modes, it seems good to include both tests.

Reply:
I guess I'd say that's *too* clever! If we ever break the "breakpoint on a line gets both locations" behavior (unlikely, I guess), we might not notice.

Reply:
Yeah, if you want to make it more explicit, then I think you would have to write:

var tinky = [1,2].map(
{$0 * 2}
)

To make sure each breakpoint takes in the appropriate context.  But I do think it is worthwhile checking those two cases as they broke in different ways.